Firestore/core/src/model/value_util.cc

```diff
@@ -486,6 +507,109 @@ std::string CanonicalId(const google_firestore_v1_ArrayValue& value) {
   return CanonifyArray(value);
 }
 
+Message<google_firestore_v1_Value> GetLowerBound(pb_size_t value_tag) {
+  switch (value_tag) {
+    case google_firestore_v1_Value_null_value_tag:
+      return NullValue();
+
+    case google_firestore_v1_Value_boolean_value_tag: {
+      Message<google_firestore_v1_Value> value;
+      value->which_value_type = value_tag;
+      value->boolean_value = false;
+      return value;
+    }
+
+    case google_firestore_v1_Value_integer_value_tag:
+    case google_firestore_v1_Value_double_value_tag: {
+      return NaNValue();
+    }
+
+    case google_firestore_v1_Value_timestamp_value_tag: {
+      Message<google_firestore_v1_Value> value;
+      value->which_value_type = value_tag;
+      value->timestamp_value.seconds = std::numeric_limits<int64_t>::min();
+      value->timestamp_value.nanos = 0;
+      return value;
+    }
+
+    case google_firestore_v1_Value_string_value_tag: {
+      Message<google_firestore_v1_Value> value;
+      value->which_value_type = value_tag;
+      value->string_value = nanopb::MakeArray<pb_bytes_array_s>(1);
```

Comment:
You should be able to set this to nullptr (and it looks like you might be leaking 1 byte here since the size is set to zero, but the allocated size is 0)